demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py

```diff
@@ -38,44 +38,72 @@ def generate_example_dict(examples_file: Optional[str], insecure=False):
     return example_dict
 
 
-def insert_outputs(yml_data: Dict, command: str, output_with_contexts: List):
-    """ Insert new ouputs for a command in the yml_data and return it.
+def insert_outputs(yml_data: Dict, command_name: str, output_with_contexts: List):
+    """ Insert new outputs for a command in the yml_data and return it.
 
     Args:
         yml_data: yaml as python dict.
-        commnad: the command we want to change the outputs of.
+        command_name: the command name whose outputs we want to manipulate.
         output_with_contexts: the new outputs.
     """
-    commands = yml_data['script']['commands']
-    found = False
-    for cmd in commands:
-        if cmd.get('name') == command:
-            cmd['outputs'] = output_with_contexts
-            found = True
-            break
-
-    if not found:
-        raise Exception(
-            f'Input YML doesn\'t have the "{command}" command that exists in the examples file.')
-
+    commands = yml_data.get('script', {}).get('commands') or []
+    command_names = [command.get('name') for command in commands]
+    if command_name not in command_names:
+        raise Exception(f'The {command_name} command is missing from the integration YML.')
+    command_index = command_names.index(command_name)
+    command = commands[command_index]
+
+    outputs: List[Dict[str, str]] = command.get('outputs') or []
+    old_descriptions = _output_path_to_description(outputs)
+    new_descriptions = _output_path_to_description(output_with_contexts)
+
+    old_output_paths: Set = set(filter(None, map(lambda val: val.get('contextPath'), command.get('outputs', []))))
```

Comment:
Do not do set from filter. Can't you use it as the iterator it wants to be?

Reply:
changed to set comprehension, think it's more readable this way.